Document GHC-45696 (#99)
haskell/error-messages: message-index/messages/GHC-45696/index.md

---
title: If-Then-Else in pattern match
summary: If-Then-Else in pattern match
severity: error
introduced: 9.6.1
---

> When pattern matching, `(if .. then .. else)` blocks are not allowed as expressions to be pattern matched against.

Reviewer: ```suggestion
When pattern matching, `(if .. then .. else)` blocks are not allowed as patterns.
```
I worry that "expressions to be pattern matched against" could be interpreted as "scrutinee"

Reviewer: Is the term "block" the right one here?

Reviewer 2: I guess instead of block we could use statement here?

Reviewer 3: s/statement/expression? :) Then I think is both clear and correct?

Reviewer: I like "expression"
